controllers/k8ssandra/k8ssandracluster_controller.go

```diff
@@ -74,6 +74,7 @@ type K8ssandraClusterReconciler struct {
 // +kubebuilder:rbac:groups=monitoring.coreos.com,namespace="k8ssandra",resources=servicemonitors,verbs=get;list;watch;create;update;patch;delete;deletecollection
 // +kubebuilder:rbac:groups=core,namespace="k8ssandra",resources=configmaps,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups="",namespace="k8ssandra",resources=events,verbs=create;patch
+// +kubebuilder:rbac:groups=batch,namespace="k8ssandra",resources=cronjobs,verbs=get;list;watch;create;update;patch;delete
```

Comment:
I've added CronJob cleanup to cleanup.go, So that will also require `list`. It also seems that `watch` is needed as without it, I see the operator logs complaining about not being able to watch CronJobs.